go/vt/vtgate/planbuilder/from.go

```diff
@@ -153,8 +154,14 @@ func (pb *primitiveBuilder) processAliasedTable(tableExpr *sqlparser.AliasedTabl
 		// a new set of column references will be generated against the new tables,
 		// and those vindex maps will be returned. They have to replace the old vindex
 		// maps of the inherited route options.
+		var tableNames []string
```

Comment:
this saves subquery original table names into "virtual" alias table. Without this, subquery tables are lost further on. The drawback of this approach  - duplicate table names appear (delete dml plan test added for this)